test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs

```diff
@@ -1321,5 +1322,57 @@ public void MsbuildRestore_WithLegacyCsproj_GlobalPackageReferencesAreProcessed(
             result.Output.Should().Contain("PackageReferences = `PackageA` / ``, `PackageB` / ``");
             result.Output.Should().Contain("PackageVersions = `PackageA` / `1.2.3`, `PackageB` / `4.5.6`");
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task MsbuildRestore_LegacyCsprojWithCpm_GlobalPackageReferencesAreProcessed(bool useStaticGraphRestore)
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+
+            // If CPM is not correctly enabled, then NuGet will see PackageReferences without a Version.
+            // By default, this is not an error condition, just a warning, but it will be limited to stable versions.
+            // By using a SemVer prerelease version, we ensure that restore always fails if CPM is not enabled (NU1103).
+            var packageX = new SimpleTestPackageContext()
+            {
+                Id = "x",
+                Version = "1.0.0-rc.1"
+            };
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageX);
+
+            var projectA = SimpleTestProjectContext.CreateLegacyPackageReference("a",
```

Comment:
Did this test fail before the product changes introduced in this PR were implemented? The reason I'm asking is that, upon examining the enum definition for [ProjectStyle](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectStyle.cs), I noticed there's no separate constant defined for legacy package reference. Therefore, I assume this project will have `ProjectStyle` set to `ProjectStyle.PackageReference`. In this case, the code should work because ProjectStyle is set to PackageReference. Please let me know if I've missed anything.

Reply:
Yes, I wrote the test before making the product change, to verify that the test was failing for static graph restore, but working for normal restore.

The MSBuild property `RestoreProjectStyle` is set by an MSBuild target `_GetRestoreProjectStyle` (which in turn is set by a task with an output parameter), which get run during "dgspec generation" during non-static graph restore. However, our static graph restore code doesn't run the `_GetRestoreProjectStyle` target. The .NET SDK explicitly sets `RestoreProjectStyle` in their props/targets, but for non-SDK style project our target usually sets it (so it can detect packages.config, PackageReference, or project.json). Therefore, our product code is currently getting a null for the value, hence why the `if` statement 2 lines up my change in the product code is allowing nulls for `projectStyle`.

I'm not 100% sure, but I think it's also related to why:
* https://github.com/NuGet/Home/issues/13109

Reply:
> However, our static graph restore code doesn't run the _GetRestoreProjectStyle target. 

I approved the PR. May the right fix is to ensure static graph restore runs that target.

Reply:
I agree we should fix reading the `RestoreProjectStyle` property in static graph restore, but the product code was still, in my opinion, a bit non-sensical. The `if` block guarding the property reads is already checking the `projectStyle` parameter, and it doesn't make sense to me why one property should have a different condition to the other properties. So, I think the product change is a good one anyway.